web/hooks/github.go

```diff
@@ -177,23 +178,35 @@ func (wh GitHubWebHook) runAssignmentTests(assignment *pb.Assignment, repo *pb.R
 
 // recordSubmissionWithoutTests saves a new submission without running any tests
 // for a manually graded assignment.
-func (wh GitHubWebHook) recordSubmissionWithoutTests(data *ci.RunData) {
+func (wh GitHubWebHook) recordSubmissionWithoutTests(rData *ci.RunData) {
+	submissionQuery := &pb.Submission{
+		AssignmentID: rData.Assignment.GetID(),
+		UserID:       rData.Repo.GetUserID(),
+		GroupID:      rData.Repo.GetGroupID(),
+	}
+	newest, err := wh.db.GetSubmission(submissionQuery)
```

Comment:
Does `newest` have the new score? It is confusing where the score comes from. Or maybe what I don't quite understand is why `recordSubmissionWithoutTests` is triggered based on a push event, rather than a gRPC call to update the benchmark/manual review.

Reply:
`newest` is the student's previous submission in the database (if any), it can have a score > 0 (set from a manual review), it can already be approved as well. However, some students push updates to the same assignment after approval, and now the submission updates, the approved status disappears and the score gets reset to zero. This is what I'm trying to fix.

`recordSubmissionWithoutTests` only triggers on a push event: it creates a submission for this lab in the database (skipping all the steps related to the tests). Only after that it is possible to create a review (a review always "belongs" to a certain submission by a remote ID).

Reply:
Added a test, but I feel like it does not belong in the `github_test.go` file since it doesn't test anything hook-related directly and is only concerned with the database behavior. Should I move it to a separate file?

Reply:
Ok. I think I now understand. I suspect that I might forget this again by the next time I look at this code, so it would be nice to have a comment in the code to explain this.

I too think the `recordSubmissionWithoutTests` function and its test doesn't belong here. Please check my recent PR #588, where I've tried to decouple `ci.RunTests` from `ci.RecordResults` (these are now methods on the `RunData` struct). I think we could do less in `ci.RecordResults` and some how make it do the same as your revised `recordSubmissionWithoutTests`, eliminating the latter function (reusing `ci.RecordResults`). We would have to move out some of the code from the this function, e.g., having to do with slip days updating and the rebuild stuff. Please take a look and let me know if there are any synergies to be had here.

Reply:
Yes, I'm on it. Found another review-related bug while testing #588 (which exists in the main, so unrelated to the PR), but the rest seems to be working fine.

Reply:
I've refactored #588 to have `ci.RecordResults` support both manual review submissions and test run submissions. I think I've updated most of the relevant code, except I haven't migrated the test case that you wrote on this PR. Could you do that and push it to #588, please. And let me know if there are any issues with my approach?

Reply:
Updated/migrated the test. No issues found.